src/code/z_collision_check.c

```diff
@@ -0,0 +1,4032 @@
+}
+
+s32 Collider_DestroyJntSphElement(GlobalContext* ctxt, ColliderJntSphElement* element) {
+    Collider_DestroyInfo(ctxt, &element->info);
+    Collider_DestroyJntSphElementDim(ctxt, &element->dim);
+    return 1;
+}
+
+s32 Collider_SetJntSphElement(GlobalContext* ctxt, ColliderJntSphElement* dest, ColliderJntSphElementInit* src) {
+    Collider_SetInfo(ctxt, &dest->info, &src->info);
+    Collider_SetJntSphElementDim(ctxt, &dest->dim, &src->dim);
+    return 1;
+}
+
+s32 Collider_ResetJntSphElementAT(GlobalContext* ctxt, ColliderJntSphElement* collider) {
+    Collider_ResetATInfo(ctxt, &collider->info);
+    return 1;
+}
+
+s32 Collider_ResetJntSphElementAC(GlobalContext* ctxt, ColliderJntSphElement* collider) {
+    Collider_ResetACInfo(ctxt, &collider->info);
+    return 1;
+}
+
+s32 Collider_ResetJntSphElementOC(GlobalContext* ctxt, ColliderJntSphElement* collider) {
+    Collider_ResetOCInfo(ctxt, &collider->info);
+    return 1;
+}
+
+/**
+ * Initializes a ColliderJntSph to default values
+ */
+s32 Collider_InitJntSph(GlobalContext* ctxt, ColliderJntSph* collider) {
+    Collider_InitBase(ctxt, &collider->base);
+    collider->count = 0;
+    collider->elements = NULL;
+    return 1;
+}
+
+/**
+ * Destroys a dynamically allocated ColliderJntSph
+ */
+s32 Collider_FreeJntSph(GlobalContext* ctxt, ColliderJntSph* collider) {
+    ColliderJntSphElement* element;
+
+    Collider_DestroyBase(ctxt, &collider->base);
+    for (element = &collider->elements[0]; element < &collider->elements[collider->count]; element++) {
+        Collider_DestroyJntSphElement(ctxt, element);
+    }
+
+    collider->count = 0;
+    if (collider->elements != NULL) {
+        zelda_free(collider->elements);
+    }
+    collider->elements = NULL;
+    return 1;
+}
+
+/**
+ * Destroys a preallocated ColliderJntSph
+ */
+s32 Collider_DestroyJntSph(GlobalContext* ctxt, ColliderJntSph* collider) {
+    ColliderJntSphElement* element;
+
+    Collider_DestroyBase(ctxt, &collider->base);
+
+    for (element = &collider->elements[0]; element < &collider->elements[collider->count]; element++) {
+        Collider_DestroyJntSphElement(ctxt, element);
+    }
+    collider->count = 0;
+    collider->elements = NULL;
+    return 1;
+}
+
+/**
+ * Sets up the ColliderJntSph using the values in src, sets it to the actor specified in src, and dynamically allocates
+ * the element array. Uses default OC2_TYPE_1 and COLTYPE_HIT0. Unused.
+ */
+s32 Collider_SetJntSphToActor(GlobalContext* ctxt, ColliderJntSph* collider, ColliderJntSphInitToActor* src) {
+    ColliderJntSphElement* destElem;
+    ColliderJntSphElementInit* srcElem;
+
+    Collider_SetBaseToActor(ctxt, &collider->base, &src->base);
+    collider->count = src->count;
+    collider->elements = zelda_malloc(src->count * sizeof(ColliderJntSphElement));
+
+    if (collider->elements == NULL) {
+        collider->count = 0;
+        return 0;
+    }
+
+    for (destElem = &collider->elements[0], srcElem = &src->elements[0];
```

Comment:
This and elsewhere does match as just `collider->elements`, I just personally prefer to note that it's an array. I don't mind changing it but I believe it can be beneficial.